source/extensions/tracers/zipkin/zipkin_tracer_impl.h

```diff
@@ -226,6 +227,8 @@ class ReporterImpl : public Reporter, Http::AsyncClient::Callbacks {
   Event::TimerPtr flush_timer_;
   const CollectorInfo collector_;
   SpanBufferPtr span_buffer_;
+  // Track active HTTP requests to be able to cancel them on destruction.
+  Http::AsyncClientRequestTracker active_requests_;
```

Comment:
qq: is this class instantiated per-thread? Or if it's shared do we need to guard with a mutex?

Reply:
yes, separate instance per thread

```
  tls_->set([this, collector, &random_generator, trace_id_128bit, shared_span_context](
                Event::Dispatcher& dispatcher) -> ThreadLocal::ThreadLocalObjectSharedPtr {
    ...
        ReporterImpl::NewInstance(std::ref(*this), std::ref(dispatcher), collector));
    ...
  });
```